src/index.ts

```diff
@@ -0,0 +1,77 @@
+import { PinejsClientCoreFactory } from 'pinejs-client-core';
+export type { PinejsClientCoreFactory } from 'pinejs-client-core';
+
+interface BackendParams {
+	/** The browser fetch API implementation or a compatible one */
+	fetch?: typeof fetch;
+}
+
+type PromiseObj = Promise<{}>;
+
+class RequestError extends Error {
+	public code = 'PineClientFetchRequestError';
+
+	constructor(
+		public body: string,
+		public status: number,
+		public responseHeaders: object,
+	) {
+		super(`Request error: ${body}`);
+	}
+}
+
+export default class PineFetch extends PinejsClientCoreFactory(Promise)<
+	PineFetch,
+	PromiseObj,
+	Promise<PinejsClientCoreFactory.PromiseResultTypes>
+> {
+	constructor(
+		params: PinejsClientCoreFactory.Params,
+		public backendParams: BackendParams,
+	) {
+		super(params);
+		if (!backendParams?.fetch && typeof fetch === 'undefined') {
+			throw new Error(
+				'No fetch implementation provided and native one not available',
+			);
+		}
+	}
+
+	_request({
+		url,
+		body,
+		...options
+	}: {
+		url: string;
+		body?: PinejsClientCoreFactory.AnyObject;
+	} & PinejsClientCoreFactory.AnyObject) {
+		const normalizedBody =
+			typeof body === 'object'
+				? JSON.stringify(body)
+				: body == null
+				? null
+				: body;
+
+		// Assign to a variable first, otherwise browser fetch errors in case the context is different.
+		const fetchImplementation = this.backendParams?.fetch ?? fetch;
+
+		return fetchImplementation(url, {
+			...options,
+			body: normalizedBody,
+		}).then((response) => {
```

Comment:
I'd use async/await for new code

Reply:
Let me try whether the transpiled umd build still works on google apps script environment :)

Reply:
Looks like it still works :+1: